### PR TITLE
Submitting changes to develop branch

### PR DESCRIPTION
**Pull request by Glubbable**
Fixed an error log spam being caused from a Sound Hook and added a Cvar to control the Menu Exit Button for fun.

### CODE REVIEW

**Comment by Starblaster64** [comment]
I always seem to forget really obvious IsValidEntity() checks, or end up making a bunch of redundant ones. Thanks for that.

The toggle menu not having an exit button was intentional. It looked nicer without the exit button when I tested it, and I didn't feel it was really necessary. If I were to add one though, I'd do it without a CVAR because there's really no point in having that as an option tbh.

If you could change it to not use a CVAR before I merge your PR that would be nice. Otherwise I'll just change it myself after the merge. (I'll merge it later after I wake up, probably)

**Comment by Glubbable** [comment]
Yeah no problem. As for the cvar option, I just added it cause I thought it'd be neat. There's other methods I could think of but I feel I would probably make something more messy than a single global cvar. So it's up to you to use, change or discard it.

**Review by Starblaster64 — APPROVED** [review]

**Pull request merged**
